### ISSUE TITLE
Warnings on int passed by reference

### ISSUE THREAD

**Issue by mmcev106** [vimeo/psalm#10074]
Does anyone have recommendations on how to avoid warnings in the specific case?

https://psalm.dev/r/6cd0ecea6a

**Comment by orklah** [comment]
not really, that's a pretty unsafe construct on its own and it would be a pain to support. 
We would need to have the equivalent of `@param` and we would also need to add it in the callable type (something like `callable(int, string) use(int): bool`)

That would solve the param issue(so the warnings from inside the callable), but not the reference(the warning after the call). We already don't support the references in callable params, so we would need to do that as well and have some syntax that allows this `callable(int -> numeric-string, string -> ?string) use(int -> bool): bool` for example. And that sort of thing is a fractal of evolutions because then people will ask for template support, conditional returns, docblock assertion support and so on...

My advice would be, if you know what you're doing, suppress all of that (or better, avoid using references :D)

**Comment by mmcev106** [comment]
Thanks very much for the feedback @orklah.
